src/Internal.AspNetCore.Sdk/targets/BuildNumber.targets

```diff
@@ -0,0 +1,37 @@
+<!--
+WARNING: These targets are indented for building Microsoft's ASP.NET Core repos, and is not intended
+for use outside of Microsoft.
+
+Usage: This file needs to be imported into the 'outer build'.
+
+For multi-targeting projects, this will be imported from buildMultiTargeting/Internal.AspNetCore.Sdk.targets.
+For single-tfm projects, this will be imported from build/Internal.AspNetCore.Sdk.targets.
+
+-->
+<Project ToolsVersion="15.0">
+  <!-- version options -->
+  <PropertyGroup>
+    <!--
+      Create second-based build number for local builds.
+      635556672000000000 is Jan 1, 2015.
+    -->
+    <_SecondBasedTimeStamp>$([System.DateTime]::UtcNow.Subtract($([System.DateTime]::FromBinary(635556672000000000))).TotalSeconds.ToString("F0"))</_SecondBasedTimeStamp>
```

Comment:
Might work. But I still prefer the shorter version.

Plus, it's possible some might build with NuGet 4.0 but need to install to project.json which uses NuGet 3.5. I'm not confident the version-length restriction is fixed there.